src/Features/CSharp/Portable/ExtractInterface/CSharpExtractInterfaceService.cs

```diff
@@ -54,15 +59,20 @@ internal override async Task<SyntaxNode> GetTypeDeclarationAsync(Document docume
             var documentWithTypeNode = solutionWithFormattedInterfaceDocument.GetDocument(documentIdWithTypeNode);
             var root = documentWithTypeNode.GetSyntaxRootSynchronously(cancellationToken);
             var typeDeclaration = root.GetAnnotatedNodes<TypeDeclarationSyntax>(typeNodeAnnotation).Single();
-
             var docId = solutionWithFormattedInterfaceDocument.GetDocument(typeDeclaration.SyntaxTree).Id;
 
-            var implementedInterfaceTypeSyntax = extractedInterfaceSymbol.TypeParameters.Any()
-                ? SyntaxFactory.GenericName(
-                    SyntaxFactory.Identifier(extractedInterfaceSymbol.Name),
-                    SyntaxFactory.TypeArgumentList(SyntaxFactory.SeparatedList(extractedInterfaceSymbol.TypeParameters.Select(p => SyntaxFactory.ParseTypeName(p.Name)))))
-                : SyntaxFactory.ParseTypeName(extractedInterfaceSymbol.Name);
+            var updatedDeclaration = UpdateTypeWithInterface(extractedInterfaceSymbol, typeNodeAnnotation, typeDeclaration);
+            var updatedRoot = root.ReplaceNode(root.GetAnnotatedNodes<TypeDeclarationSyntax>(typeNodeAnnotation).Single(), updatedDeclaration);
+            var solutionWithOriginalTypeUpdated = solutionWithFormattedInterfaceDocument.WithDocumentSyntaxRoot(docId, updatedRoot, PreservationMode.PreserveIdentity);
+            return solutionWithOriginalTypeUpdated;
```

Comment:
please wrap line longer than github's width (around 120 column is a good default).